src/NativeScript/GlobalObject.mm

```diff
@@ -118,6 +119,27 @@ static void microtaskRunLoopSourcePerformWork(void* context) {
     self->drainMicrotasks();
 }
 
+static void runLoopBeforeWaitingPerformWork(CFRunLoopObserverRef observer, CFRunLoopActivity activity, void* info) {
```

Comment:
Is this function supposed to be executed before the processing of the run loop? If this is the case, is it worth it to check that the activity is an entry one.